pkg/hydrator/batch.go

```diff
@@ -0,0 +1,155 @@
+package hydrator
+
+import (
+	"context"
+
+	"github.com/asecurityteam/nexpose-vuln-hydrator/pkg/domain"
+)
+
+// BatchSolutionFetcher represents an interface for concurrently fetching solutions
+type BatchSolutionFetcher interface {
```

Comment:
I realize these shouldn't be exported but I was having a tough time figuring out naming for these.

Reply:
I don't really have a good suggestion for this problem. Naming things is hard.

As long as they aren't exported, we can play loose with the typical naming scheme. For example, it's not uncommon in other languages to prefix interfaces with `i` such as `ibatchSolutionFetcher` or to suffix implementations with `impl` such as `batchSolutionFetcherImpl`. I think these may be "java-isms" that most Go devs would dislike very much but if it's not exported then it shouldn't really matter to a consumer of the project.

I'm not sure if all of these can actually be unexported, though. Anything that the `Hydrator` implementation needs to work will need to be exported so that `main.go` can construct one.

All of this is to say that things might be fine the way they are.